server/image_list_libpod.go

```diff
@@ -0,0 +1,64 @@
+// +build libpod
+
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+// ListImages lists existing images.
+func (s *Server) ListImages(ctx context.Context, req *pb.ListImagesRequest) (resp *pb.ListImagesResponse, err error) {
+	const operation = "list_images"
+	defer func() {
+		recordOperation(operation, time.Now())
+		recordError(operation, err)
+	}()
+
+	logrus.Debugf("ListImagesRequest: %+v", req)
+	filter := ""
+	reqFilter := req.GetFilter()
+	if reqFilter != nil {
+		filterImage := reqFilter.GetImage()
+		if filterImage != nil {
+			filter = filterImage.Image
+		}
+	}
+
+	// TODO fix the filtering.
+	_ = filter
```

Comment:
I'm surprised integration tests pass w/o filtering, we should add these tests if we don't have them (not in this PR).
I'd love to fix this before merging I guess, this support is broken otherwise Kube-wise